src/ui/Snackbar/SnackbarContext.tsx

```diff
@@ -0,0 +1,65 @@
+import { ItemType } from '@common-types';
+import { nanoid } from 'nanoid';
+import React, {
+  ComponentProps,
+  createContext,
+  ReactNode,
+  useContext,
+  useMemo,
+} from 'react';
+import { useQueue } from 'src/hooks/useQueue';
+import SnackbarList from './SnackbarList';
+
+type TSnackbar = ItemType<ComponentProps<typeof SnackbarList>['snacks']>;
+type TSnackbarContext = {
+  enqueue(alert: Pick<TSnackbar, 'severity' | 'message'>): void;
+  showError(message: ReactNode): void;
+  showSuccess(message: ReactNode): void;
```

Comment:
Let's add a `showWarning` for completeness here

Reply:
There's also info, I put `showError` and `showSuccess` because they are common. Other cases can trigger manually via `enqueue`. Or, we can do it for all 4 types and remove enqueue from context.

Reply:
I think let's do it for all 4 types but I do see the benefit of the `enqueue()`.

This method should be of type `enqueue(alert: Pick<TSnackbar, 'id' | 'severity' | 'message'>): void;` which allow passing in of a custom ID (in which case we won't generate the nanoid random ID but use this instead). This ID can be used in a `close()` method which allow dimissing of the snackbar from outside.

Reply:
You mean like this?
```ts
const {close} = useSnackbar();
close(id);
```
Can you give me one use case for this? Else it just adds complexity to the component.

Reply:
Can't think of a good use case yet... Just see that every snackbar lib has this feature. But it's okay as is. We can easily add it in when we need it.